oci/archive/oci_transport.go

```diff
@@ -0,0 +1,117 @@
+package archive
+
+import (
+	"github.com/containers/image/docker/reference"
+	"github.com/containers/image/image"
+	"github.com/containers/image/oci/layout"
+	"github.com/containers/image/transports"
+	"github.com/containers/image/types"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	transports.Register(Transport)
+}
+
+// Transport is an ImageTransport for OCI archive
+// it creates an oci-archive tar file by calling into the OCI transport
+// tarring the directory created by oci and deleting the directory
+var Transport = ociArchiveTransport{}
+
+type ociArchiveTransport struct{}
+
+// ociArchiveReference is an ImageReference for OCI Archive paths
+type ociArchiveReference struct {
+	dir         string
+	resolvedDir string
+	tag         string
+	ociRef      types.ImageReference
+}
+
+func (t ociArchiveTransport) Name() string {
+	return "oci-archive"
+}
+
+// ParseReference converts a string, which should not start with the ImageTransport.Name prefix
+// into an ImageReference.
+func (t ociArchiveTransport) ParseReference(reference string) (types.ImageReference, error) {
+	// ociArchRef stores the dir, resolvedDir, and tag of the original path given by the user
+	ociArchRef, err := layout.Transport.ParseReference(reference)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error parsing reference %q", reference)
+	}
+	dir, resolvedDir, tag := layout.GetOciReference(ociArchRef)
```

Comment:
This is still passing a file path to a transport which expects a directory.

Please just copy&paste most of `oci/layout/oci_transport.go` into this file, more or less with `s/dir/file/g; s/layout/archive/g` instead of bending the semantics like this.  Or perhaps create a separate helper subpackage for the shared code, which is explicitly documented to use generic paths without knowing whether they are files or directories.